pkg/intermediate/aggregate.go

```diff
@@ -645,9 +694,124 @@ func (a *AggregationProcess) addFieldsForStatsAggregation(record entities.Record
+	if err != nil {
+		return err
+	}
+	reverseByteCount, err = getUnsigned64ValueByIeName(record, "reverseOctetTotalCount")
+	if err != nil {
+		return err
+	}
+	incomingVal := byteCount / (uint64(timeEnd - timeStart))
+	reverseIncomingVal := reverseByteCount / (uint64(timeEnd - timeStart))
+	throughputVals := []uint64{incomingVal, reverseIncomingVal}
+
+	for i, element := range antreaThroughputElements {
+		// add common throughput elements
+		value := throughputVals[i]
+		ie, err := registry.GetInfoElement(element, registry.AntreaEnterpriseID)
+		if err != nil {
+			return err
+		}
+		if err = record.AddInfoElement(entities.NewUnsigned64InfoElement(ie, value)); err != nil {
+			return err
+		}
+		// add source throughput elements
+		value = uint64(0)
+		ie, err = registry.GetInfoElement(antreaSourceThroughputElements[i], registry.AntreaEnterpriseID)
+		if err != nil {
+			return err
+		}
+		if fillSrcStats {
+			value = throughputVals[i]
+		}
+		if err = record.AddInfoElement(entities.NewUnsigned64InfoElement(ie, value)); err != nil {
+			return err
+		}
+		// add destination throughput elements
+		value = uint64(0)
+		ie, err = registry.GetInfoElement(antreaDestinationThroughputElements[i], registry.AntreaEnterpriseID)
+		if err != nil {
+			return err
+		}
+		if fillDstStats {
+			value = throughputVals[i]
+		}
+		if err = record.AddInfoElement(entities.NewUnsigned64InfoElement(ie, value)); err != nil {
+			return err
+		}
+	}
+
+	// Initialize the elements flowEndSecondsFromSource/DestinationNode.
+	if timeEnd != uint32(0) {
+		var srcTime, dstTime uint32
+		if fillSrcStats {
+			srcTime = timeEnd
+		}
+		if fillDstStats {
+			dstTime = timeEnd
+		}
+		ie, err := registry.GetInfoElement("flowEndSecondsFromSourceNode", registry.AntreaEnterpriseID)
+		if err != nil {
+			return err
+		}
+		if err = record.AddInfoElement(entities.NewUnsigned32InfoElement(ie, srcTime)); err != nil {
+			return err
+		}
+		ie, err = registry.GetInfoElement("flowEndSecondsFromDestinationNode", registry.AntreaEnterpriseID)
+		if err != nil {
+			return err
+		}
+		if err = record.AddInfoElement(entities.NewUnsigned32InfoElement(ie, dstTime)); err != nil {
+			return err
+		}
+	}
 	return nil
 }
 
+func (a *AggregationProcess) updateFlowEndSecondsFromNodes(incomingRecord, existingRecord entities.Record, ieName string, ieVal uint32) uint32 {
+	existingIe, _, _ := existingRecord.GetInfoElementWithValue(ieName)
+	existingVal := existingIe.GetUnsigned32Value()
+	// When the incoming record is the first record from one node, the previous record
+	// doesn't exist. time_diff = flowEndSeconds - flowStartSeconds
+	if existingVal == 0 {
+		incomingIe, _, _ := incomingRecord.GetInfoElementWithValue("flowStartSeconds")
+		existingVal = incomingIe.GetUnsigned32Value()
+	}
+	if existingVal < ieVal {
+		existingIe.SetUnsigned32Value(ieVal)
+	}
+	return existingVal
+}
+
+func getUnsigned64ValueByIeName(record entities.Record, ieName string) (uint64, error) {
```

Comment:
we can consider adding similar methods for record interface in the future

Reply:
marked as a TODO